src/events/messageCreate.ts

```diff
@@ -1,27 +1,61 @@
 import {
-	Message,
 	MessageOptions,
 	MessageActionRow,
 	MessageButton,
 	ThreadChannel,
+	TextChannel,
 } from 'discord.js';
 import { event } from 'jellycommands';
-import url_regex from 'url-regex-safe';
 import { AUTO_THREAD_CHANNELS } from '../config';
 import { wrap_in_embed } from '../utils/embed_helpers';
 import { add_thread_prefix } from '../utils/threads';
-import { get_title_from_url } from '../utils/unfurl';
-import { Url } from 'url';
 
 export default event({
 	name: 'messageCreate',
-	run: async ({}, message) => {
+	run: async ({ }, message) => {
+		// Rules for whether or not the message should be dealt with by the bot
 		const should_ignore =
 			message.author.bot ||
 			message.channel.type != 'GUILD_TEXT' ||
 			message.type != 'DEFAULT' ||
 			!AUTO_THREAD_CHANNELS.includes(message.channelId);
 
+		// If a response is sent by a user in an auto thread channel
+		if (message.type === 'REPLY' && AUTO_THREAD_CHANNELS.includes(message.channelId) && !message.author.bot) {
+			// Delete their message
+			await message.delete()
+			// Get the channel as a TextChannel
+			const channel = message.channel as TextChannel
```

Comment:
Should a check be put here to make sure it can be cast?

Reply:
It probably should, but based on where it is it can't really ever be anything but a TextChannel, so it shouldn't be possible for it to fail at that point